src/js/components/Form/Form.js

```diff
@@ -536,8 +560,21 @@ const Form = forwardRef(
               validateArg,
               required,
             );
+
+            // priority is given to validateOn prop on formField, if it is
+            // undefined, then we will use the validate prop value of Form.
+            // The reason we don't want to add validateOn = "submit" here is
+            // because we don't have to trigger validation of "submit" field
```

Comment:
```suggestion
            // The reason we don't want to add validateOn = "submit" here is
            // because we don't want to trigger validation of "submit" field
```